### PR TITLE
Add project requirements + drag & drop

### PR DESCRIPTION
**Pull request by Ewoskyx**
### To-Do App 

In this project I did;
- Add All project requirements like add, remove, update from local storage according to index and indexes also generated according to array length of local storage.
- The index of the items is re-arranged if any item or items get deleted from the task list.
- Try the OPTIONAL requirement Drag & Drop, currently it has a minor bug which I will handle in the next feature update.

### CODE REVIEW

File: src/app/ui.js [hunk 1]
```diff
@@ -35,6 +38,11 @@ class UI {
     });
   }
 
+  static reloadUI() {
+    const childs = dynamicDiv.querySelectorAll('.task-main');
+    childs.forEach((child) => child.remove());
+  }
+
   static clearTask() {
```
Comment (bimbolabuari) [inline src/app/ui.js:51]: Kindly use an array.filter method to remove task from the array ✊

File: src/index.js [hunk 2]
```diff
@@ -40,6 +40,11 @@ form.addEventListener('keyup', (e) => {
...
-  // eslint-disable-next-line no-restricted-globals
-  location.reload();
+  UI.reloadUI();
+  UI.showTasks();
 });
+
+// Drag&Drop
+UI.reloadUI();
+UI.showTasks();
+UI.dragOver();
```
Comment (bimbolabuari) [inline src/index.js:50]: Kindly add the 'clear all completed' function using the filter method as required 🙂

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @Ewoskyx , 

You have done a good job so far 🚀  

## Highlights 🎉 

- Implement a function for adding task
- Implement a function for removing task.

There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

File: src/app/ui.js (hunk 1 above)
Comment (Ewoskyx) [reply src/app/ui.js:51]: Hello Abimbola, According to the code review, the part you have highlighted is not removing any specific tasks. Instead that part is removing whole tasks and adds again just for the use of DOM update.

 However, as project required, the filter function is already applied you can see at  " delete selected tasks " in the storage class in deleteFromStore() method. Could you please check it once again if you have time. Thank you in advance and thanks for the review :)

File: src/app/ui.js [hunk 3]
```diff
@@ -49,6 +57,40 @@ class UI {
...
+
+  static dragUI(e) {
+    e.target.classList.add('drag');
+  }
+
+  static stopDrag(e) {
+    e.target.classList.remove('drag');
+  }
+
+  static dragOver() {
```
Comment (carlos-ssh) [inline src/app/ui.js:74]: # 💾 💾 💽 [OPTIONAL]   💽 💾 💾
- When i try to move a task at the top of the list i can't do it

File: src/app/ui.js [hunk 4]
```diff
@@ -35,6 +38,11 @@ class UI {
     });
   }
 
+  static reloadUI() {
+    const childs = dynamicDiv.querySelectorAll('.task-main');
+    childs.forEach((child) => child.remove());
```
Comment (carlos-ssh) [inline src/app/ui.js:48]: Kindly use `filter()` to remove your items.

- it's requested on your project requirements.
Screen Shot 2021-12-04 at 8 03 13

File: src/app/ui.js [hunk 5]
```diff
@@ -49,6 +57,40 @@ class UI {
     const newTask = document.querySelector('.task-input');
     newTask.value = '';
   }
```
Comment (carlos-ssh) [inline src/app/ui.js:64]: Kindly fix this issue:

- Add a function for remove single `items` from the array requested in your `project requirements`:


Screen Shot 2021-12-04 at 8 07 49


You have a `function` for removes selected items.
Try to follow all the requirements requested before to request another code review.

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
# 🦺 🚧 🚥 CHANGES REQUIRED 🚥 🚧 🦺
#

Hi @Ewoskyx , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._ 


changesrequested
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app/ui.js (hunk 4 above)
Comment (Ewoskyx) [reply src/app/ui.js:48]: Thank you for the review. But, once again the same. Let me explain again then. The requirement says "Clear all COMPLETED" that method you highlight removes EVERYTHING, not the completed ones!
Please check it once again or explain to me the need for filtering if I want to remove EVERYTHING even it's completed or not that method is removing ALL ITEMS and RESTORES. 
However, the clear all completed method is still there. Please kindly check the STORE class and the method 
filter

So the clear all completed button is USING the method deleteFromStore() not the reloadUI()! reloadUI is just reloads the DOM...

File: src/app/ui.js (hunk 5 above)
Comment (Ewoskyx) [reply src/app/ui.js:64]: change
And this is the part for editing the task description. Please check again or shall I send a video that shows everything in action?
You already can remove selected ITEMS or SINGLE ITEM depending on your selection. That method is also deleteFromSTore().

**Review by carlos-ssh — APPROVED** [review]
# 🍾🍾 **STATUS APPROVED** 👏🏽👏🏽

Hi @Ewoskyx, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Approved


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**